### DISCUSSION TITLE
Telegram组件中要怎么设置Bot的快捷指令?

### DISCUSSION THREAD

**Discussion by Flock-YT** [simple-robot/Discussions#192] [Q&A] [answered]
### 涉及的编程语言

Java

### 涉及的组件库

Telegram

### 疑问描述

Telegram的官方API中有一个setMyCommands API用于设置Bot的快捷指令,请问在Simbot的Telegram组件中该怎么调用API设置快捷指令呢?

相关示例图:
![image](https://github.com/user-attachments/assets/9800b9f7-a34a-4cc9-80ea-58e07abf3a5b)

**Comment by ForteScarlet** [comment]
目前仅实现了 [GetMyCommandsApi](https://docs.simbot.forte.love/components/telegram/simbot-component-telegram-api/love.forte.simbot.telegram.api.bot.command/-get-my-commands-api/index.html) ，尚未实现 SetMyCommands API

**Reply by Flock-YT** [reply]
了解

**Reply by Flock-YT** [reply]
注意到在v0.0.8版本中已经实现了SetMyCommands API,但是我在参考了示例以及结合以前开发QQ机器人的经验后还是没搞懂要怎么调用,构建好API后找不到类似api.requestData的方法,请问是已经有新的调用方式了吗?
![image](https://github.com/user-attachments/assets/dfc81447-3038-4e06-af65-a4c9507d89ae)

**Reply by ForteScarlet** [reply, marked as answer]
可参考 [#21](https://github.com/simple-robot/simbot-component-telegram/pull/21) 的代码示例直接使用 `TelegramBot` 中的API，如果想要直接使用原始API，Java中需要通过静态类 `TelegramBotRequests` 使用（`TelegramBotRequests.requestxxx` 之类的）

当然，示例是Kotlin的，在Java中的样子可能会有些差异

指正：`TelegramBotRequests` 在 JVM 平台的API尚未实现，直接用 `TelegramBot` 中的 `commands` API吧

**Reply by Flock-YT** [reply]
丢一个JAVA的示例在这里以防有人要用到

```
        // 从Application对象中获取Bot管理器
        BotManagers botManagers = application.getBotManagers();

        // 获取所有的Bot，并将其存储在一个序列中
        Sequence<Bot> botSequence = botManagers.allBots();

        // 获取Bot序列的迭代器
        Iterator<Bot> iterator = botSequence.iterator();

        // 遍历迭代器中的每个Bot
        while (iterator.hasNext()) {
            // 获取下一个Bot并转换成TelegramBot类型
            TelegramBot bot = (TelegramBot) iterator.next();

            // 获取命令更新器并添加命令 阻塞线程
            bot.commandsBlocking()
                    .getUpdater()
                    .addCommand("start", "开始使用Bot")
                    .addCommand("help", "获取帮助信息")
                    .addCommand("info", "查看Bot信息")
                    .updateBlocking();


        }
```
